searchcore/src/vespa/searchcore/proton/reprocessing/reprocess_documents_task.cpp

```diff
@@ -30,55 +28,40 @@ ReprocessDocumentsTask(IReprocessingInitializer &initializer,
     }
 }
 
-
 void
 ReprocessDocumentsTask::run()
 {
     if (_handler.hasProcessors()) {
-        EventLogger::reprocessDocumentsStart(_subDbName,
-                                             _visitorCost);
-        fastos::TimeStamp ts(fastos::ClockSystem::now());
-        _startTime = ts.ms();
-        _loggedTime = _startTime;
+        EventLogger::reprocessDocumentsStart(_subDbName,_visitorCost);
+        _stopWatch.start();
         search::IDocumentStore &docstore = _sm->getBackingStore();
         if (_handler.hasRewriters()) {
-            docstore.accept(_handler.getRewriteVisitor(),
-                    *this,
-                    *_docTypeRepo);
+            docstore.accept(_handler.getRewriteVisitor(),*this,*_docTypeRepo);
         } else {
-            docstore.accept(_handler,
-                    *this,
-                    *_docTypeRepo);
+            docstore.accept(_handler,*this,*_docTypeRepo);
         }
         _handler.done();
-        ts = fastos::ClockSystem::now();
-        int64_t elapsedTime = ts.ms() - _startTime;
-        EventLogger::reprocessDocumentsComplete(_subDbName,
-                                                _visitorCost,
-                                                elapsedTime);
+        _stopWatch.stop();
+        EventLogger::reprocessDocumentsComplete(_subDbName,_visitorCost, _stopWatch.elapsed().ms());
     }
 }
 
-
 void
 ReprocessDocumentsTask::updateProgress(double progress)
 {
     _visitorProgress = progress;
     double deltaProgress = progress - _loggedProgress;
     if (deltaProgress >= 0.01) {
-        fastos::TimeStamp ts = fastos::ClockSystem::now();
-        int64_t logDelayTime = ts.ms() - _loggedTime;
-        if (logDelayTime >= 60000 || deltaProgress >= 0.10) {
-            EventLogger::reprocessDocumentsProgress(_subDbName,
-                                                    progress,
-                                                    _visitorCost);
-            _loggedTime = ts.ms();
+        fastos::StopWatch intermediate = _stopWatch;
+        fastos::TimeStamp logDelayTime = intermediate.stop().elapsed() - _stopWatch.elapsed();
```

Comment:
This is a bit hard to understand. It presumably relies on `elapsed()` implicitly returning zero if the stopwatch has not been stopped yet?

Reply:
Or more correctly previous previous "mellomtid".